### PR TITLE
fix: add checking if there is data in the base table for long window deploy

### LINKED ISSUE #2483: tablet keeps CPU 160%
**Bug Description**
version：0.6.1
(gdb) bt
#0  0x00007f155879f5dd in nanosleep () from /lib64/libc.so.6
#1  0x00007f15587cfde4 in usleep () from /lib64/libc.so.6
#2  0x0000000001bcd137 in bthread_usleep (microseconds=microseconds@entry=1000000) at /__w/hybridsql-asserts/hybridsql-asserts/deps/build/src/brpc/src/bthread/bthread.cpp:363
#3  0x0000000001cb0afa in brpc::Server::RunUntilAskedToQuit (this=this@entry=0x7ffe42314240) at /__w/hybridsql-asserts/hybridsql-asserts/deps/build/src/brpc/src/brpc/server.cpp:1700
#4  0x00000000005ae7d7 in StartTablet() () at /__w/OpenMLDB/OpenMLDB/src/cmd/openmldb.cc:268
#5  0x000000000051c5c7 in main (argc=<optimized out>, argv=<optimized out>) at /__w/OpenMLDB/OpenMLDB/src/cmd/openmldb.cc:3945
(gdb) quit
A debugging session is active.

	Inferior 1 [process 26558] will be detached.

Quit anyway? (y or n) y
Detaching from program: /opt/huawei/data/OnlineLearning/openmldb-tablet-0.6.1/bin/openmldb, process 26558

重复持续打印以下日志
W0916 17:10:01.019234 26583 log_reader.cc:529] no log part matched! start_offset[0]
W0916 17:10:01.019245 26574 log_reader.cc:529] no log part matched! start_offset[0]
W0916 17:10:01.019256 26583 log_reader.cc:477] fail to roll read log
W0916 17:10:01.019268 26574 log_reader.cc:529] no log part matched! start_offset[0]
W0916 17:10:01.019277 26583 log_reader.cc:529] no log part matched! start_offset[0]
W0916 17:10:01.019289 26574 log_reader.cc:477] fail to roll read log
W0916 17:10:01.019299 26583 log_reader.cc:529] no log part matched! start_offset[0]
W0916 17:10:01.019307 26574 log_reader.cc:529] no log part matched! start_offset[0]
W0916 17:10:01.019318 26583 log_reader.cc:477] fail to roll read log
W0916 17:10:01.019330 26574 log_reader.cc:529] no log part matched! start_offset[0]
W0916 17:10:01.019340 26583 log_reader.cc:529] no log part matched! start_offset[0]
W0916 17:10:01.019351 26574 log_reader.cc:477] fail to roll read log
W0916 17:10:01.019362 26583 log_reader.cc:529] no log part matched! start_offset[0]
W0916 17:10:01.019372 26574 log_reader.cc:529] no log part matched! start_offset[0]
W0916 17:10:01.019382 26583 log_reader.cc:477] fail to roll read log
W0916 17:10:01.019392 26574 log_reader.cc:529] no log part matched! start_offset[0]
W0916 17:10:01.019402 26583 log_reader.cc:529] no log part matched! start_offset[0]
W0916 17:10:01.019413 26574 log_reader.cc:477] fail to roll read log
W0916 17:10:01.019424 26583 log_reader.cc:529] no log part matched! start_offset[0]
W0916 17:10:01.019435 26574 log_reader.cc:529] no log part matched! start_offset[0]
W0916 17:10:01.019445 26583 log_reader.cc:477] fail to roll read log
W0916 17:10:01.019455 26574 log_reader.cc:529] no log part matched! start_offset[0]
W0916 17:10:01.019466 26583 log_reader.cc:529] no log part matched! start_offset[0]
W0916 17:10:01.019477 26574 log_reader.cc:477] fail to roll read log
W0916 17:10:01.019488 26583 log_reader.cc:529] no log part matched! start_offset[0]
W0916 17:10:01.019500 26574 log_reader.cc:529] no log part matched! start_offset[0]
W0916 17:10:01.019510 26583 log_reader.cc:477] fail to roll read log
W0916 17:10:01.019521 26574 log_reader.cc:529] no log part matched! start_offset[0]
W0916 17:10:01.019532 26583 log_reader.cc:529] no log part matched! start_offset[0]
W0916 17:10:01.019542 26574 log_reader.cc:477] fail to roll read log
W0916 17:10:01.019552 26583 log_reader.cc:529] no log part matched! start_offset[0]

**Expected Behavior**

**Steps to Reproduce**

1. 
2. 
3. 
4.

### PR DESCRIPTION
**Pull request by zhanghaohit**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Fix #2483

- add checking if there is data in the base table for long window deployment
- add checking in LogReader for `offset`

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

### CODE REVIEW

File: src/storage/aggregator.cc [hunk 1]
```diff
@@ -321,12 +324,16 @@ bool Aggregator::Init(std::shared_ptr<LogReplicator> base_replicator) {
         }
         it->NextPK();
     }
-    if (aggr_table_->GetRecordCnt() == 0) {
-        recovery_offset = 0;
+    if (recovery_offset == 0 || recovery_offset == UINT64_MAX) {
+        PDLOG(INFO, "aggregator recovery failed, recovery_offset=%lu", recovery_offset);
```
Comment (dl239) [inline src/storage/aggregator.cc]: WARNING

File: src/cmd/sql_cmd.h [hunk 2]
```diff
@@ -42,6 +42,7 @@ DEFINE_string(spark_conf, "", "The config file of Spark job");
 DECLARE_string(zk_cluster);
 DECLARE_string(zk_root_path);
 DECLARE_int32(zk_session_timeout);
+DECLARE_uint32(get_table_status_interval);
```
Comment (dl239) [inline src/cmd/sql_cmd.h]: this gflags has no effect in sdk

Comment (zhanghaohit) [reply src/cmd/sql_cmd.h]: oh. yes. I should move it to the `sql_cmd_test.cc`

**Review by dl239 — APPROVED** [review]

File: src/replica/log_replicator_test.cc [hunk 3]
```diff
@@ -147,6 +150,78 @@ TEST_F(LogReplicatorTest, BenchMark) {
     ASSERT_TRUE(ok);
 }
 
+TEST_F(LogReplicatorTest, LogReader) {
+    // set to 1 MB, every binlog file will be a little larger than 2 MB
+    // as the checking logic is: (wh_->GetSize() / (1024 * 1024)) > (uint32_t)FLAGS_binlog_single_file_max_size
+    FLAGS_binlog_single_file_max_size = 1;
+    std::map<std::string, std::string> map;
+    std::string folder = "/tmp/" + GenRand() + "/";
```
Comment (aceforeverd) [inline src/replica/log_replicator_test.cc]: `/tmp` -> `std::filesystem::temp_directory_path()`

File: src/cmd/sql_cmd_test.cc [hunk 4]
```diff
@@ -860,6 +830,37 @@ class DeployLongWindowEnv {
...
+
+    std::string deploy_sql = "deploy test_aggr options(LONG_WINDOWS='w1:2') select col1, col2,"
+        " sum(i64_col) over w1 as w1_sum_i64_col,"
+        " from " + base_table +
+        " WINDOW w1 AS (PARTITION BY " + base_table + ".col1," + base_table + ".col2 ORDER BY col3"
+        " ROWS_RANGE BETWEEN 5 PRECEDING AND CURRENT ROW);";
+    sr->ExecuteSQL(base_db, "use " + base_db + ";", &status);
+    ASSERT_TRUE(status.IsOK()) << status.msg;
+    sr->ExecuteSQL(base_db, deploy_sql, &status);
+    ASSERT_TRUE(!status.IsOK()) << status.msg;
```
Comment (aceforeverd) [inline src/cmd/sql_cmd_test.cc]: assert the msg ?

Comment (zhanghaohit) [reply src/cmd/sql_cmd_test.cc]: In this case, it should deploy failed.

**Review by vagetablechicken — APPROVED** [review]

**Review by aceforeverd — APPROVED** [review]

**Pull request merged**